src/SensorManager.php

```diff
@@ -96,7 +142,7 @@ public function stage(ExecutionStage $executionStage): void
 
     public function request(Request $request, Response $response): void
     {
-        $sensor = new RequestSensor(
+        $sensor = $this->requestSensor ??= new RequestSensor(
```

Comment:
We now capture thee request and command sensor as properties which allows us to switch those out in our tests.